### ISSUE TITLE
PlantUML Fenced Code is Not Rendered

### ISSUE THREAD

**Issue by kidomine** [vsch/idea-multimarkdown#740]
PlantUML diagrams are not rendered on my setup (see below).
Bundled Markdown plugin for PyCharm is disabled.
I also have a separate PlantUML Integration plugin which works fine on its own.

I also tried the following:
* disabled both PlantUML Integration and bundled Markdown plugin - still NG
* replaced plantuml.jar of Markdown Navigator with the one that came with PlantUML Integration - also NG

I also observed this in another computer (Windows 10, same PyCharm version).

System Configuration:
* OS: macOS Mojave
* PyCharm CE - 2019.1.1
* Markdown Navigator - v2.8.4
* PlantUML Integration - v2.14.3

Is this feature supposed to be for enhanced version of Markdown Navigator only?

**Comment by vsch** [comment]
@kidomine, plant UML integration is part of the basic plugin and does not require the licensed version.

There is a bug in the released version that does not save a change in puml language parser setting. You either need to change another setting in addition to the PlantUML extension checkbox so that the settings page detects a change in setting.

Alternately, you can use the EAP channel and install a fixed plugin.

EAP update with a fix for the issue is available.

[Installing EAP Plugin Version](https://github.com/vsch/idea-multimarkdown/wiki/EAP-Update-Channel)

**Comment by kidomine** [comment]
@vsch Thanks for the quick response. I tried EAP as you suggested, and I am now using v2.8.4.16.
However, I still can't render PlantUML diagrams in fenced code.

I also tried the following configurations, all of which failed (i.e. PlantUML is not rendered):
1. changed `Markdown->Preview->HTML Generator`: Swing vs JavaFx
2. changed `Markdown->Preview->HTML Generation->HTML Generator`: Swing vs JavaFx
3. changed `Markdown->Preview->HTML Generation->PlantUML Rendering`: Embedded Library vs Gravizo SVG/PNG

I also observed that while the `Markdown Copy Action` is enabled in the preview pane, the dropdown items are all disabled (e.g. clicking on `Export Rendered HTML`, etc. does not do anything).

This seems to be an issue in the Preview Pane itself.
Is there any way we can narrow down the issue?

If there are additional info that you need, just let me know.

Below are my sample tests and the corresponding preview.
<pre>
```puml
class A {
   +field1: string
}
```
```plantuml
class B {
   +field1: string
}
```
~~~puml
class C {
   +field1: string
}
~~~
~~~plantuml
class D {
   +field1: string
}
~~~
</pre>

<img width="1138" alt="Screen Shot 2019-04-17 at 8 57 33 PM" src="https://user-images.githubusercontent.com/4011927/56289445-8438b700-6153-11e9-8025-ba624c386ad6.png">

**Comment by vsch** [comment]
@kidomine, I am using the same configuration as you have described, including PlantUML plugin which I have installed in all my iDEs, and I cannot duplicate the issue. The only difference is that I am running on PyCharm not PyCharm CE. I will try the latter to see if there is a difference.

<img width="538" alt="image" src="https://user-images.githubusercontent.com/10299086/56323053-b4269100-6138-11e9-820d-2db697646a87.png">

Both Swing and JavaFX browsers work.

The following settings are used, with relevant settings highlighted:

![image](https://user-images.githubusercontent.com/10299086/56323407-be955a80-6139-11e9-8cf2-5d10e0908b53.png)

![image](https://user-images.githubusercontent.com/10299086/56323449-e7b5eb00-6139-11e9-8fdb-5c1cc3643dc2.png)

**Comment by vsch** [comment]
@kidomine, the export and copy functionality is only available in the enhanced edition so the items for these actions are disabled in the basic edition.

**Comment by vsch** [comment]
@kidomine, same for PyCharm CE. Renders correctly.

The image cache is located in the user home directory under `~/.markdownNavigator/image-cache` it should include all images generated or converted from SVG. When running the IDE can you see this directory?

**Comment by kidomine** [comment]
@vsch It seems the culprit are the last checkboxes under `Parser` (i.e. `XXX fenced code` settings). It seems these are disabled by default, which gave me an impression that the rendering does not work.

My environment is OK now, and I already switched back to official release instead of EAP. I did not try to downgrade to the previous official release, but I think it should still be OK.

Re the export functionality, I understand.
Also, `~/.markdownNavigator/image-cache` seems fine, I can see the PNG contents there now.

Thanks for the quick responses!

I will close this issue now.
